File: rc-react-native-rtc/android/src/main/java/cn/rongcloud/rn/rtc/RCReactNativeRtcViewManager.java
package cn.rongcloud.rn.rtc;

import androidx.annotation.NonNull;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

import cn.rongcloud.rtc.wrapper.RCRTCIWView;
import cn.rongcloud.rtc.wrapper.constants.RCRTCIWViewFitType;

public class RCReactNativeRtcViewManager extends SimpleViewManager<RCRTCIWView> {
    @NonNull
    @Override
    public String getName() {
        return "RCReactNativeRtcView";
    }

    @NonNull
    @Override
    protected RCRTCIWView createViewInstance(@NonNull ThemedReactContext context) {
        return new RCRTCIWView(context);
    }

    @ReactProp(name = "fitType")
    public void setFitType(RCRTCIWView view, int fitType) {
        view.setFitType(RCRTCIWViewFitType.values()[fitType]);
    }

    @ReactProp(name = "mirror")
    public void setMirror(RCRTCIWView view, boolean mirror) {
        view.setMirror(mirror);
    }
}

File: ios/RCReactNativeRtcViewManager.m
//
//  RCReactNativeRtcViewManager.m
//  RCReactNativeRtc
//
//  Created by 潘铭达 on 2021/7/13.
//

#import "RCReactNativeRtcViewManager.h"
#import <RongRTCLibWrapper/RongRTCLibWrapper.h>

@implementation RCReactNativeRtcViewManager

RCT_EXPORT_MODULE(RCReactNativeRtcView)

RCT_EXPORT_VIEW_PROPERTY(fitType, int);

RCT_EXPORT_VIEW_PROPERTY(mirror, BOOL);

- (UIView *)view {
    RCRTCIWView *view = [RCRTCIWView create];
    return view;
}

@end
